core/src/main/java/com/novoda/noplayer/exoplayer/forwarder/BitrateForwarder.java

```diff
@@ -0,0 +1,57 @@
+package com.novoda.noplayer.exoplayer.forwarder;
+
+import com.google.android.exoplayer2.C;
+import com.google.android.exoplayer2.Format;
+import com.google.android.exoplayer2.source.AdaptiveMediaSourceEventListener;
+import com.google.android.exoplayer2.upstream.DataSpec;
+import com.novoda.noplayer.listeners.BitrateChangedListeners;
+
+import java.io.IOException;
+
+class BitrateForwarder implements AdaptiveMediaSourceEventListener {
+
+    private Bitrate videoBitrate = Bitrate.fromBitsPerSecond(0);  //Check if same logic in MediaPlayer
+    private Bitrate audioBitrate = Bitrate.fromBitsPerSecond(0);
+
+    private final BitrateChangedListeners bitrateChangedListeners;
+
+    BitrateForwarder(BitrateChangedListeners bitrateChangedListeners) {
+        this.bitrateChangedListeners = bitrateChangedListeners;
+    }
+
+    @Override
+    public void onDownstreamFormatChanged(int trackType, Format trackFormat, int trackSelectionReason, Object trackSelectionData, long mediaTimeMs) {
+        if (trackType == C.TRACK_TYPE_VIDEO) {
+            videoBitrate = Bitrate.fromBitsPerSecond(trackFormat.bitrate);
+            bitrateChangedListeners.onBitrateChanged(audioBitrate, videoBitrate);
+        } else if (trackType == C.TRACK_TYPE_AUDIO) {
+            audioBitrate = Bitrate.fromBitsPerSecond(trackFormat.bitrate);
+            bitrateChangedListeners.onBitrateChanged(audioBitrate, videoBitrate);
+        }
+    }
+
+    @Override
+    public void onLoadStarted(DataSpec dataSpec, int dataType, int trackType, Format trackFormat, int trackSelectionReason, Object trackSelectionData, long mediaStartTimeMs, long mediaEndTimeMs, long elapsedRealtimeMs) {
+        //TODO should we send ?
```

Comment:
These `TODOs` were hidden in the abstract `ExoPlayerEventForwarder` class, not anymore! We should figure out what to do with these!